internal/lang/eval.go

```diff
@@ -462,6 +468,21 @@ func (s *Scope) evalContext(refs []*addrs.Reference, selfAddr addrs.Referenceabl
 	return ctx, diags
 }
 
+func getRunBlockValue(refs []*addrs.Reference, outputValues map[string]cty.Value) map[string]cty.Value {
```

Comment:
Like my previous comment, this logic shouldn't be here, as it leaks into the general tofu logic. The `evalContext` is the general context for the plan/apply, and you are populating it with something strictly related to the test's feature. It shouldn't be here.
In this [PR](https://github.com/opentofu/opentofu/pull/1129/files#diff-16f7ccd0b34c619fda941531b649e919dcc79d87306854db9d212c3050296305R1063-R1075) @cam72cam  already implemented what you are trying to do in this function - save the run block outputs. As part of his PR, he passed the `run` outputs so other `run` blocks would be able to access them.
So IMO, you need to use his existing logic and figure out how to use the run block outputs in the `provider` block inside the test file.

Reply:
Thanks @ronnyorot , I will check the code again and figure out the solution.